netbench-orchestrator/src/ec2_utils/instance.rs

```diff
@@ -0,0 +1,211 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    ec2_utils::{
+        launch_plan::LaunchPlan,
+        types::{Az, EndpointType, HostIps, PrivIp, PubIp},
+    },
+    orchestrator::{HostConfig, OrchError, OrchResult, OrchestratorConfig, STATE},
+};
+use aws_sdk_ec2::types::{
+    BlockDeviceMapping, EbsBlockDevice, IamInstanceProfileSpecification, Instance,
+    InstanceNetworkInterfaceSpecification, InstanceStateName, InstanceType, PlacementGroup,
+    ResourceType, ShutdownBehavior, Tag, TagSpecification,
+};
+use std::{collections::HashMap, net::IpAddr, str::FromStr, time::Duration};
+use tracing::info;
+
+pub async fn launch_instances(
+    ec2_client: &aws_sdk_ec2::Client,
+    launch_plan: &LaunchPlan<'_>,
+    security_group_id: &str,
+    unique_id: &str,
```

Comment:
Its created [here](https://github.com/aws/s2n-netbench/pull/26/files#diff-d3bec5ed67122f6cf207d65b0100354523d1b7ebe9b360700ac54b9bc1b4fa35R20-R24). The idea is to create a unique identifier for the netbench runs since multiple parties might launch a request at the same time.

This is used for constructing the report url and transparent to the user.

Reply:
The `unique_id` is created once in main and threaded thru the entire process.